sycl/test/warnings/warnings.cpp

```diff
@@ -1,17 +1,42 @@
 // RUN: %clangxx -fsycl --no-system-header-prefix=CL/sycl -fsyntax-only -Wall -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wpessimizing-move -Wunused-variable -Wmismatched-tags -Wunneeded-internal-declaration -Werror -Wno-unknown-cuda-version -Wno-unused-parameter %s -o %t.out
-
+// RUN: %clangxx -fsycl -E --no-system-header-prefix=CL/sycl %s -o %t.ii
+// RUN: %clangxx -fsycl -fsyntax-only -Wall -Wextra -Wno-ignored-attributes -Wno-deprecated-declarations -Wpessimizing-move -Wunused-variable -Wmismatched-tags -Wunneeded-internal-declaration -Werror -Wno-unknown-cuda-version -Wno-unused-parameter %t.ii -o %t.out
```

Comment:
BTW, I don't know why `-Wno-unused-parameter` was added here, but we can pass this test just fine without disabling that diagnostic

Reply:
Outlined this into #3980